rancher2/schema_cluster_aks_config.go

```diff
@@ -270,11 +277,55 @@ func clusterAKSConfigFields() map[string]*schema.Schema {
 			Description: "A CIDR notation IP range from which to assign Kubernetes Service cluster IPs. It must not overlap with any Subnet IP ranges",
 		},
 		"tag": &schema.Schema{
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Computed:    true,
+			Description: "Tags for Kubernetes cluster. For example, foo=bar. Deprecated, use `tags` argument instead",
+		},
+		"tags": &schema.Schema{
 			Type:        schema.TypeMap,
 			Optional:    true,
 			Computed:    true,
 			Description: "Tags for Kubernetes cluster. For example, foo=bar",
 		},
+		"enable_auto_scaling": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Enable auto-scaling of worker nodes",
+		},
+		"min_count": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Default:     1,
+			Description: "Minimum number of worker nodes in the auto-scaling cluster",
+		},
+		"max_count": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Default:     10,
+			Description: "Maximum number of worker nodes in the auto-scaling cluster",
+		},
+		"agent_pool_type": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "AvailabilitySet",
+			Description: "Agent pool type (VirtualMachineScaleSets or AvailabilitySet).",
+		},
+		"availability_zones": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "List of availability zones to use for the cluster",
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"load_balancer_sku": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "Basic",
```

Comment:
As discussed [here](https://github.com/rancher/kontainer-engine/pull/200#discussion_r351636524), value should be "basic" instead of "Basic". Adding validation function 

```
-			Default:     "Basic",
+			Default:     "basic",
+                       ValidateFunc: validation.StringInSlice(clusterAKSLoadBalancerSku, true),
```